### PR TITLE
WIP: Pr/cargo fixes

### PR DESCRIPTION
**Pull request by hunger**
Some moe mechanical cleanup work: This time to make clippy useful by getting rid of noise.

The patches with "Janitor" prefix should not change the behavior.

Those without might actually fix bugs.

### CODE REVIEW

File: xtask/src/license_headers_check.rs [hunk 1]
```diff
@@ -324,8 +324,8 @@ impl CargoDependency {
             toml_edit::Value::InlineTable(table) => {
                 if let (Some(path), Some(version)) = (table.get("path"), table.get("version")) {
                     Some(Self::Full {
-                        path: path.as_str().unwrap_or_default().clone().into(),
-                        version: version.as_str().unwrap_or_default().clone().into(),
+                        path: *(path.as_str().unwrap_or_default()).clone().into(),
+                        version: *(version.as_str().unwrap_or_default()).clone().into(),
```
Comment (ogoffart) [inline xtask/src/license_headers_check.rs:328]: I wonder if the clone is actually necessary here

File: sixtyfps_compiler/passes/binding_analysis.rs [hunk 2]
```diff
@@ -23,7 +23,7 @@ use crate::object_tree::{Component, ElementRc};
 
 type PropertySet = linked_hash_set::LinkedHashSet<NamedReference>;
 
-pub fn binding_analysis(component: &Rc<Component>, diag: &mut BuildDiagnostics) -> () {
+pub fn binding_analysis(component: &Rc<Component>, diag: &mut BuildDiagnostics) {
```
Comment (ogoffart) [inline sixtyfps_compiler/passes/binding_analysis.rs:26]: ( I blame rust analyser for adding these when creating functions 😉)

File: sixtyfps_runtime/corelib/animations.rs [hunk 3]
```diff
@@ -63,7 +63,7 @@ impl core::ops::AddAssign<core::time::Duration> for Instant {
 
 impl core::ops::SubAssign<core::time::Duration> for Instant {
     fn sub_assign(&mut self, other: core::time::Duration) {
-        self.0 += other.as_millis() as u64;
+        self.0 -= other.as_millis() as u64;
```
Comment (ogoffart) [inline sixtyfps_runtime/corelib/animations.rs:66]: Duh!

Comment (tronical) [inline sixtyfps_runtime/corelib/animations.rs:66]: Oops. I hope this was dead code :)

File: sixtyfps_compiler/passes/remove_aliases.rs [hunk 4]
```diff
@@ -184,7 +184,7 @@ pub fn remove_aliases(component: &Rc<Component>, diag: &mut BuildDiagnostics) {
                         .or_default()
                         .is_set = true;
                 } else {
-                    drop(d);
+                    drop(std::mem::take(d));
```
Comment (ogoffart) [inline sixtyfps_compiler/passes/remove_aliases.rs:187]: This is wrong.
If this compile the drop of d can be removed. (but not the drop of elem)
mem::take is wrong

Comment (hunger) [reply sixtyfps_compiler/passes/remove_aliases.rs:187]: I dropped this drop.

**Comment by ogoffart** [comment]
What's the status of this PR?  
Did you mive patches to other PR?

**Comment by hunger** [comment]
I am trying to get this merged piece by piece.

I think I'll just close this one?

**Pull request closed without merging**